### PR TITLE
Add Windows.Detection.Yara.NTFS

### PR DESCRIPTION
**Pull request by mgreen27**
Add Windows.Detection.Yara.NTFS and test.
 This artifact searches the MFT, returns a list of target files then runs Yara  over the target list.

### CODE REVIEW

File: artifacts/testdata/server/testcases/yara_detection.out.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          LastModified0x30 = "2021-01-24T06:57:39.5353986Z",
+          LastRecordChange0x10 = "2021-01-24T06:59:40.7021579Z",
+          LastRecordChange0x30 = "2021-01-24T06:57:39.5353986Z",
+          LastAccess0x10 = "2021-03-05T13:44:04.1999087Z",
+          LastAccess0x30 = "2021-01-24T06:57:36.7073284Z")
+      ])
+FROM scope()
+[]SELECT FullPath,Size,Rule,Rule,Meta,HitContext,HitOffset FROM Artifact.Windows.Detection.Yara.NTFS(PathRegex='wkscli.dll',ShortHandYara='wide ascii:MZ')[
+ {
+  "FullPath": "d:\\a\\velociraptor\\velociraptor\\artifacts/testdata/files/wkscli.dll",
```
Comment (scudette) [inline artifacts/testdata/server/testcases/yara_detection.out.yaml:31]: We need to remove the leading directory here or it wont be able to run locally. Try to pass here 

`rel(path=FullPath, base=srdDir)`

Comment (mgreen27) [reply artifacts/testdata/server/testcases/yara_detection.out.yaml:31]: Local works ok as my mocked MFT result is: MFTPath = srcDir + "$MFT".
The FullPath is calculated with: split(string=MFTPath,sep='\\$')[0] + FullPath as FullPath

File: artifacts/definitions/Windows/Detection/Yara/NTFS.yaml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      -- check which Yara to use
+      LET yara = SELECT * FROM if(condition=YaraUrl,
+            then= { SELECT Content FROM http_client( url=YaraUrl, method='GET') },
+            else= if(condition=ShortHandYara,
+                then= { SELECT ShortHandYara as Content FROM scope() },
+                else= { SELECT YaraRule as Content FROM scope() }))
+            
+      -- first find all matching files mft
+      LET files = SELECT IsDir,
+            split(string=MFTPath,sep='\\$')[0] +  FullPath as FullPath
```
Comment (scudette) [inline artifacts/definitions/Windows/Detection/Yara/NTFS.yaml:79]: Is this still needed? Shouldn't we just use the drive letter here?

Comment (mgreen27) [reply artifacts/definitions/Windows/Detection/Yara/NTFS.yaml:79]: This is still required as if the user used the AllDrives option the only way to determine the Drive is with the split function.

**Pull request merged**